pkgs/os-specific/linux/dcgm/default.nix

```diff
@@ -116,14 +108,18 @@ in gcc11Stdenv.mkDerivation rec {
   ];
 
   buildInputs = [
-    plog.dev # header-only
-    tclap_1_4 # header-only
-
+    # Header-only
     catch2
-    fmt_9
-    jsoncpp-static
-    libevent-nossl-static
-    yaml-cpp
+    plog.dev
+    tclap_1_4
+
+    # Dependencies that can be either static or dynamic.
+    (fmt_9.override { enableShared = !static; }) # DCGM's build uses the static outputs regardless of enableShared
+    (yaml-cpp.override { inherit static; stdenv = gcc11Stdenv; })
```

Comment:
Why do we have to override `stdenv`? If we use `gcc11Stdenv` for `yaml-cpp` but not for the other packages it'll be linked to a different `libc` (when linked dynamically)